tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsSQLTestsBaseTrait.scala

```diff
@@ -121,6 +121,12 @@ object RapidsSQLTestsBaseTrait {
         "org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback")
       .set("spark.sql.warehouse.dir", warehouse)
       .set("spark.sql.cache.serializer", "com.nvidia.spark.ParquetCachedBatchSerializer")
+      .set("spark.sql.session.timeZone", "UTC")
```

Comment:
Do we really want to force UTC in the tests?  We have partial support for timezones and it's likely more will be added later.  Typically what we've done in the past is leave it up to the environment setup to specify the timezone desired to be tested (e.g.: CI scripts will set the TZ environment variable or other JVM settings to setup the timezone) rather than have the tests smash the timezone directly.  That way the tests can be run across multiple timezones by changing the environment before running the tests.

Comment applies to a couple of other places below where the timezone is being smashed with UTC.

Reply:
the consideration of spark.sql.session.timeZone is explained in previous comment

Reply:
@GaryShen2008 told me that they're already suggesting our customer to set timezone to UTC to facilitate GPU execution. So at least we're not testing a fictitious scenario

Reply:
We should enable non-UTC test cases later, but in the first step, setting UTC here is to enable us testing in a fixed config which we support now. It can help us to detect the failure in UTC case. Without this UTC setting, the test cases may pass because of fallback to CPU. Currently we haven't been able to detect the wrong fallback in UT.

Reply:
I'm not sure why we're not setting UTC in the _environment_ of the test setup, which lets us reuse these tests.  If we're going to leave this as-is, there needs to be a TODO comment linking to a tracking issue to fix this.

Reply:
I'll put it into test environment setup script